### ISSUE TITLE
Lerna attempts to publish private packages

### ISSUE THREAD

**Issue by ianwremmel** [lerna/lerna#286]
I'm pretty sure 1.x would ignore packages with `private:true` in package.json, but 2.x spits out the following:

```
npm ERR! This package has been marked as private
npm ERR! Remove the 'private' field from the package.json to publish it.
```

**Comment by ianwremmel** [comment]
It looks like it's specific to --canary releases. 

collectUpdates() has a filter condition that's basically `<complex logic> || true`

**Comment by gigabo** [comment]
Hi @ianwremmel, #321 landed in `lerna@2.0.0-beta.28`.  That should get publish with private modules working correctly.  Please reopen (or start a new ticket) if you still run into trouble after upgrading.
